craft_parts/plugins/rust_plugin.py

```diff
@@ -14,36 +14,50 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-"""The rust plugin."""
+"""The craft Rust plugin."""
 
 import logging
+import subprocess
+from textwrap import dedent
 from typing import TYPE_CHECKING, Any, Dict, List, Optional, Set, cast
 
 from overrides import override
-from pydantic import conlist
+from pydantic import conlist, validator
 
-from . import validator
 from .base import Plugin, PluginModel, extract_plugin_properties
 from .properties import PluginProperties
 
 logger = logging.getLogger(__name__)
 
 # A workaround for mypy false positives
 # see https://github.com/samuelcolvin/pydantic/issues/975#issuecomment-551147305
+# The proper fix requires Python 3.9+ (needs `typing.Annotated`)
 if TYPE_CHECKING:
     UniqueStrList = List[str]
 else:
     UniqueStrList = conlist(str, unique_items=True)
 
 
 class RustPluginProperties(PluginProperties, PluginModel):
-    """The part properties used by the rust plugin."""
+    """The part properties used by the Rust plugin."""
 
     # part properties required by the plugin
+    rust_deps: UniqueStrList = []
     rust_features: UniqueStrList = []
     rust_path: UniqueStrList = ["."]
+    rust_channel: Optional[str] = None
+    rust_use_global_lto: bool = False
+    rust_no_default_features: bool = False
     source: str
 
+    @validator("rust_channel")
+    @classmethod
+    def check_rust_deps(cls, value: Optional[str], values) -> Optional[str]:
+        """Validate that system Rust and rustup Rust are not used together."""
+        if value and values.get("rust_deps"):
```

Comment:
```yaml
after = values.get("after")
if value and "rust-deps" in after
```

Reply:
This is not possible. `after` is not passed to this part.